All Category game count problem

@claude 

I'm having an ongoing issue with the All category and infinite scrolling on the front end. When I get to the bottom of the page after scrolling for a bunch, it stops with 'scroll for more' 208 of 220 or something like that.

It doesn't scroll anymore though. 
Can you try and find the root cause of this?
I'm not sure if its because all the games are showing that should show but its not counting correctly at the bottom?
220 games is correct for the total amount of games

@claude 

getting a testing error:
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/pages/__tests__/PublicCatalogue.test.jsx > PublicCatalogue Page > Infinite scroll > loads more games when scrolling near bottom
AssertionError: expected "vi.fn()" to be called at least once
 ❯ src/pages/__tests__/PublicCatalogue.test.jsx:500:43
    498| 
    499|       // Verify IntersectionObserver was set up
    500|       expect(global.IntersectionObserver).toHaveBeenCalled();
       |                                           ^
    501|     });
    502|   });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

--- a/frontend/src/pages/__tests__/PublicCatalogue.test.jsx
+++ b/frontend/src/pages/__tests__/PublicCatalogue.test.jsx
@@ -207,10 +207,16 @@ describe('PublicCatalogue Page', () => {
       expect(screen.getByText('Game 1')).toBeInTheDocument();
     });
 
+    // Wait for the load more trigger element to be present
+    await waitFor(() => {
+      const loadMoreText = screen.queryByText(/scroll for more/i);
+      expect(loadMoreText).toBeInTheDocument();
+    });
+
     // Verify IntersectionObserver was created (wait for useEffect to complete)
     await waitFor(() => {
       expect(global.IntersectionObserver).toHaveBeenCalled();
-    });
+    }, { timeout: 2000 });
   });
 
   test('displays load more indicator when more pages available', async () => {
@@ -492,12 +498,21 @@ describe('PublicCatalogue Page', () => {
         </BrowserRouter>
       );
 
+      // Wait for the games to load first
       await waitFor(() => {
         expect(screen.getByText('Game 1')).toBeInTheDocument();
       });
 
-      // Verify IntersectionObserver was set up
-      expect(global.IntersectionObserver).toHaveBeenCalled();
+      // Wait for the load more trigger element to be present (needed for intersection observer)
+      await waitFor(() => {
+        const loadMoreText = screen.queryByText(/scroll for more/i);
+        expect(loadMoreText).toBeInTheDocument();
+      });
+
+      // Give the intersection observer useEffect time to run after the ref is attached
+      await waitFor(() => {
+        expect(global.IntersectionObserver).toHaveBeenCalled();
+      }, { timeout: 2000 });
     });
   });
 

